### PR TITLE
[backport #10398] Give a better error message for external shares with self-signed ssl cer...

### LINKED ISSUE #13850: Can't add a shared link to the cloud the link was shared from
The ownCloud Android app only lets you share a file through a link. Naturally, you would assume that to share a file with somebody else on your cloud, you send them the link. They can use the 'add this file to your owncloud' button to, well, add it to their cloud. Unfortunately, this gives an error as ownCloud does not detect you are already both on the same installation.

Of course, the Android (and probably iOS) client can be fixed in this regard but it seems something that should work.

### Steps to reproduce
1. Share a link to a user on the same cloud as you
2. User opens link and adds it to the ownCloud instance you both use
3. ownCloud gives an error: can't add shared link

### Expected behaviour
ownCloud detects you're on the same server and just adds the file as if it was shared by user name.

### Actual behaviour
ownCloud gives an error.

### Server configuration
ownCloud 7 EE on S3 ;-)

@schiesbn with oC8 this might be relatively easy to do - Junior Job maybe?

### PR DESCRIPTION
**Pull request by DeepDiver1975**
...tificates

fixes #13850

@karlitschek @schiesbn @jospoortvliet

### CODE REVIEW

**Comment by karlitschek** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
@owncloud-bot retest this please

**Comment by MorrisJobke** [comment]
https://github.com/owncloud/core/pull/13880/files?w=1

**Comment by MorrisJobke** [comment]
Works :+1:

**Comment by jospoortvliet** [comment]
Great work all!

File: apps/files_sharing/ajax/external.php [hunk 1]
```diff
@@ -39,16 +39,22 @@
...
-/**
- * @var \OCA\Files_Sharing\External\Storage $storage
- */
-$storage = $mount->getStorage();
-$result = $storage->file_exists('');
-if($result){
-	$storage->getScanner()->scanAll();
-	\OCP\JSON::success();
+// check for ssl cert
+if (substr($remote, 0, 5) === 'https' and !OC_Util::getUrlContent($remote)) {
```
Comment (schiesbn) [inline apps/files_sharing/ajax/external.php]: ```OC_Util::getUrlContent($remote)``` not only fails if the ssl certificate is invalid but also for any other network error, right? So there is a chance that we send the wrong error message, all network errors are now displayed as ssl certificate error. But it is probably still better than having no warning about potential certificate problems?

Comment (DeepDiver1975) [inline apps/files_sharing/ajax/external.php]: > Ideally we could parse the response message and always display the correct error.

sounds like an enhancment for 8.1 - please open a new ticket - THX

**Pull request merged**